include/libcyphal/presentation/publisher.hpp

```diff
@@ -0,0 +1,207 @@
+    using Failure = transport::AnyFailure;
+
+    PublisherBase(const PublisherBase& other)
+        : impl_{other.impl_}
+        , priority_{other.priority_}
+    {
+        CETL_DEBUG_ASSERT(impl_ != nullptr, "Not supposed to copy from already moved `other`.");
+        impl_->retain();
+    }
+
+    PublisherBase(PublisherBase&& other) noexcept
+        : impl_{std::exchange(other.impl_, nullptr)}
+        , priority_{other.priority_}
+    {
+        CETL_DEBUG_ASSERT(impl_ != nullptr, "Not supposed to move from already moved `other`.");
+    }
+
+    PublisherBase& operator=(const PublisherBase& other)
+    {
+        if (this != &other)
+        {
+            CETL_DEBUG_ASSERT(impl_ != nullptr, "Not supposed to copy to already moved `this`.");
+            CETL_DEBUG_ASSERT(other.impl_ != nullptr, "Not supposed to copy from already moved `other`.");
+
+            impl_->release();
+
+            impl_     = other.impl_;
+            priority_ = other.priority_;
+
+            impl_->retain();
+        }
+        return *this;
+    }
+
+    PublisherBase& operator=(PublisherBase&& other) noexcept
+    {
+        CETL_DEBUG_ASSERT(impl_ != nullptr, "Not supposed to move to already moved `this`.");
+        CETL_DEBUG_ASSERT(other.impl_ != nullptr, "Not supposed to move from already moved `other`.");
+
+        impl_->release();
+
+        impl_     = std::exchange(other.impl_, nullptr);
+        priority_ = other.priority_;
+
+        return *this;
+    }
+
+    transport::Priority getPriority() const noexcept
+    {
+        return priority_;
+    }
+
+    void setPriority(const transport::Priority priority) noexcept
+    {
+        priority_ = priority;
+    }
+
+protected:
+    ~PublisherBase() noexcept
+    {
+        if (impl_ != nullptr)
+        {
+            impl_->release();
+        }
+    }
+
+    explicit PublisherBase(PublisherImpl* const impl)
+        : impl_{impl}
+        , priority_{transport::Priority::Nominal}
+    {
+        CETL_DEBUG_ASSERT(impl != nullptr, "");
+
+        impl->retain();
+    }
+
+    cetl::optional<Failure> publishRawData(const TimePoint deadline, const cetl::span<const cetl::byte> data) const
+    {
+        CETL_DEBUG_ASSERT(impl_ != nullptr, "");
+
+        return impl_->publishRawData(deadline, priority_, data);
+    }
+
+private:
+    // MARK: Data members:
+
+    PublisherImpl*      impl_;
+    transport::Priority priority_;
+
+};  // PublisherBase
+
+}  // namespace detail
+
+// TODO: docs
+template <typename Message>
+class Publisher final : public detail::PublisherBase
+{
+    static_assert(!Message::_traits_::IsServiceType, "Service types are not supported by the Publisher.");
+
+public:
+    using Failure = libcyphal::detail::AppendType<transport::AnyFailure, nunavut::support::Error>::Result;
+
+    cetl::optional<Failure> publish(const TimePoint deadline, const Message& message) const
+    {
+        std::array<std::uint8_t, Message::_traits_::SerializationBufferSizeBytes> buffer{};
```

Comment:
Zero-initialization is prohibitively slow for large messages and must be avoided.

```suggestion
        std::array<std::uint8_t, Message::_traits_::SerializationBufferSizeBytes> buffer;
```

Reply:
It was actually like this (without `{}`), but later in my unit "deserialization failure" tests I've got once unexpected/random scenario, so I decided to zero it. I though it is fine b/c we want (afaik) anyway eliminate this copying in the first place (by somehow "teaching" our  `bitspan` to work with ScatteredBuffer directly).